sphinx_gallery/gen_rst.py

```diff
@@ -438,21 +438,35 @@ def execute_script(code_block, example_globals, image_path, fig_count,
 
         # Depending on whether we have one or more figures, we're using a
         # horizontal list or a single rst call to 'image'.
+        image_list = ""
         if len(figure_list) == 1:
             figure_name = figure_list[0]
             image_list = SINGLE_IMAGE % figure_name.lstrip('/')
-        else:
+        elif len(figure_list) > 1:
             image_list = HLIST_HEADER
             for figure_name in figure_list:
                 image_list += HLIST_IMAGE_TEMPLATE % figure_name.lstrip('/')
 
+
     except Exception:
-        figure_list = []
-        image_list = '%s is not compiling:' % src_file
+        formatted_exception = traceback.format_exc()
+
         print(80 * '_')
-        print(image_list)
-        traceback.print_exc()
+        print('%s is not compiling:' % src_file)
+        print(formatted_exception)
         print(80 * '_')
+
+        figure_list = []
+        image_list = codestr2rst(formatted_exception, 'pytb')
```

Comment:
For readability, I would suggest being explicit about the keyword argument here, by writing:

<pre>
    ..., lang='pytb')
</pre>